bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -266,14 +266,29 @@ impl Client {
 impl Client {
     /// Information about login options for the client's homeserver.
     pub async fn homeserver_login_details(&self) -> Arc<HomeserverLoginDetails> {
-        let supports_oidc_login = self.inner.oidc().fetch_authentication_issuer().await.is_ok();
+        let oidc = self.inner.oidc();
+        let (supports_oidc_login, supports_oidc_registration) =
+            if let Ok(issuer) = oidc.fetch_authentication_issuer().await {
+                if let Ok(metadata) = &oidc.given_provider_metadata(&issuer).await {
+                    metadata
+                        .prompt_values_supported
```

Comment:
Wouldn't be more efficient / generic to expose the `prompt_values_supported` list as a list of `OidcPrompt`?

Reply:
Sure, I wasn't sure if the concept of `OidcPrompt`s was a bit too much info but if we're happy with that I'll update and at which point, there wouldn't be any custom FFI logic so I would leave all this in the FFI.

Reply:
Hmmm, the amount of code didn't really change that much, it's basically just the use of `Into` instead of `contains`. WDYT @bnjbvr, I wouldn't really consider there to be any custom FFI logic here now as it is just mapping SDK values to FFI values.

Reply:
If that's sufficient for your needs, that's fine with me. (Please make sure to log errors that are silently swallowed here, for both `if let Ok(...)` branches.)

Reply:
Done in https://github.com/matrix-org/matrix-rust-sdk/commit/4829f48ce609f1809143217f5cddb7172378933d